gui/Graph.cpp

```diff
@@ -51,20 +43,34 @@ void Graph::draw() {
 			SDL_RenderDrawLine(this->renderer, currentX, currentY, nextX, nextY);
 		}
 		// TODO: if the objects are not functions, we have to draw a line between
-		// the last and the first point as well!
+		// the last and the first point as well: if (mode == "objects")
 	}
 
 	SDL_SetRenderDrawColor(this->renderer, 0, 0, 0, 255);
 }
 
+void Graph::addObject(Object *obj) {
+	debug("adding an object\n\n");
+	this->objects.push_back(obj);
+	int x = obj->getPoints()[0].x();
+	int y = obj->getPoints()[0].y();
+	debug(std::to_string(x));
+	debug(std::to_string(y));
+	debug("\n\n");
+}
+
 void Graph::setObjects(Object *obj) {
 	std::vector<Object*> emptyVector;
 	this->objects = emptyVector;
 	this->objects.push_back(obj);
 }
```

Comment:
We either need a function that takes `Point` and makes a small circle out of it or we have to do it before calling the function, maybe we should add an issue for "Point as Object"-support?

Reply:
My suggestion would be that when the user enters "P(1/1)", we treat it as if he has entered "C(1/1, <x>)", with x being a fixed radius. That way, we only have to handle the "conversion" from Point to Circle in the Parser to make an Object. Either way, this should not be handled here, but in the Parser, I think. In the Graph.objects vector, we never should have a Point in my opinion.

Reply:
That works for me!